### PR TITLE
feat: [Novo] Convert /search route

### PR DESCRIPTION
**Pull request by damassi**
Addresses: https://artsyproduct.atlassian.net/browse/PLATFORM-3122 

Moves `/search` page over to novo.

### CODE REVIEW

File: src/v2/Artsy/Router/Utils/shouldUpdateScroll.tsx [hunk 1]
```diff
@@ -7,7 +7,7 @@
 export function shouldUpdateScroll(_prevRenderArgs, { routes }) {
   try {
     // If true, don't reset scroll position on route change
-    if (routes.some(route => route.ignoreScrollBehavior)) {
+    if (routes?.some(route => route.ignoreScrollBehavior)) {
```
Comment (damassi) [inline src/v2/Artsy/Router/Utils/shouldUpdateScroll.tsx:10]: This has masked errors in the past since it sits so low in the stack

File: src/common-middleware.ts [hunk 2]
```diff
@@ -157,4 +158,5 @@ export default function commonMiddlewareSetup(app) {
    * global router.
    */
   app.get("/artist/*", artistMiddleware)
+  app.get("/search*", searchMiddleware)
```
Comment (joeyAghion) [inline src/common-middleware.ts:161]: Since [I think] this is scoped to the search route, could it just happen within that app instead of common middleware?

File: src/v2/Apps/Artist/Routes/Consign/Components/Subheader.tsx [hunk 3]
```diff
@@ -12,9 +12,9 @@ export const Subheader: React.FC<SubheaderProps> = ({
 }) => {
   return (
     <Box maxWidth="80%" m="auto">
-      <Serif element="h2" size="10" color={color}>
+      <Text as="h2" variant="title" color={color}>
         {children}
-      </Serif>
+      </Text>
```
Comment (joeyAghion) [inline src/v2/Apps/Artist/Routes/Consign/Components/Subheader.tsx:17]: I presume this is unrelated. Is it just a best-practice/clean-up, or does it impact the page?

File: src/v2/Apps/getAppNovoRoutes.tsx [hunk 4]
```diff
@@ -4,10 +4,12 @@ import { artistRoutes } from "v2/Apps/Artist/artistRoutes"
 import { artistSeriesRoutes } from "./ArtistSeries/artistSeriesRoutes"
 import { artistsRoutes } from "v2/Apps/Artists/artistsRoutes"
 import { artworkRoutes } from "v2/Apps/Artwork/artworkRoutes"
+import { auctionsRoutes } from "./Auctions/auctionsRoutes"
 import { collectRoutes } from "v2/Apps/Collect/collectRoutes"
 import { consignRoutes } from "v2/Apps/Consign/consignRoutes"
 import { conversationRoutes } from "v2/Apps/Conversation/conversationRoutes"
 import { debugRoutes } from "./Debug/debugRoutes"
+import { exampleRoutes } from "./Example/exampleRoutes"
```
Comment (joeyAghion) [inline src/v2/Apps/getAppNovoRoutes.tsx:12]: Good call, especially if this is used as the basis of new routes.

**Review by joeyAghion — COMMENTED** [review]
We usually have to update [the list of `convertedPaths` in `webpackPublicPath.ts`](https://github.com/artsy/force/blob/17000ceaed06e93a520241f29b901ffe65d017b7/src/desktop/lib/webpackPublicPath.ts#L26-L41) as well. Is that missing from this PR?

File: src/v2/Apps/getAppNovoRoutes.tsx (hunk 4 above)
Comment (icirellik) [reply src/v2/Apps/getAppNovoRoutes.tsx:12]: If this truly is an example it shouldn't be deployed live in production. I also feel the same about the baseline page, it doesn't provide any real value in production either.

File: src/common-middleware.ts (hunk 2 above)
Comment (icirellik) [reply src/common-middleware.ts:161]: Since it appears to be doing a redirect having it in a middleware would be the more performant solution.

File: src/lib/middleware/searchMiddleware.tsx [hunk 5]
```diff
@@ -0,0 +1,25 @@
+import { getContextPageFromReq } from "lib/getContextPage"
+import { OwnerType } from "@artsy/cohesion"
+import { stringify } from "qs"
+
+export const searchMiddleware = async (req, res, next) => {
+  const { pageType } = getContextPageFromReq(req)
+
+  if (pageType === OwnerType.search) {
```
Comment (icirellik) [inline src/lib/middleware/searchMiddleware.tsx:8]: Now that this route is scope to the search page in express, the check should be removed.

**Review by icirellik — COMMENTED** [review]
A few minor changes. I would prefer to see search in its own PR because it has multiple secondary changes and converting several routes here will make it much more difficult to debug and rollback.

Additionally, have you visually diff'd the pages to ensure the css is correct?

**Comment by icirellik** [comment]
Also, Calibre Routes should be added for each route that is missing. If there are new routes please do a couple manual snapshots before merging this PR.

**Comment by icirellik** [comment]
You'll also need to update `webpackPublicPath.ts`

File: src/v2/Apps/getAppNovoRoutes.tsx (hunk 4 above)
Comment (damassi) [reply src/v2/Apps/getAppNovoRoutes.tsx:12]: I can follow up on that in a separate PR; we have an admin only flag I need to port over.

File: src/v2/Apps/Artist/Routes/Consign/Components/Subheader.tsx (hunk 3 above)
Comment (damassi) [reply src/v2/Apps/Artist/Routes/Consign/Components/Subheader.tsx:17]: I noticed there was some font weirdness looking at the page, and the old Serif component has been deprecated.

**Comment by damassi** [comment]
@icirellik Address all remaining feedback

**Review by joeyAghion — APPROVED** [review]

**Review by icirellik — APPROVED** [review]
